cmd/agent/common/misconfig/global.go

```diff
@@ -8,19 +8,33 @@ package misconfig
 import "github.com/DataDog/datadog-agent/pkg/util/log"
 
 // ToLog outputs warnings about common misconfigurations in the logs
-func ToLog() {
+func ToLog(agent AgentType) {
 	for name, check := range checks {
-		if err := check(); err != nil {
-			log.Warnf("misconfig: %s: %v", name, err)
+		if _, ok := check.supportedAgents[agent]; ok {
+			if err := check.run(); err != nil {
+				log.Warnf("misconfig: %s: %v", name, err)
+			}
 		}
 	}
 }
 
+type AgentType int
 type checkFn func() error
+type check struct {
+	name            string
+	run             checkFn
+	supportedAgents map[AgentType]struct{}
+}
+
+const (
+	CoreAgent = iota
+	ProcessAgent
+)
 
-var checks = map[string]checkFn{}
+var checks = map[string]check{}
 
 // nolint: deadcode, unused
 func registerCheck(name string, c checkFn) {
-	checks[name] = c
+	supportedAgentsSet := map[AgentType]struct{}{CoreAgent: {}, ProcessAgent: {}}
+	checks[name] = check{name: name, run: c, supportedAgents: supportedAgentsSet}
```

Comment:
Note that not all agents support the `CoreAgent` and the `ProcessAgent`. I would pass `supportedAgentsSet` into `registerCheck` as a function parameter

Reply:
so when `registerCheck` gets called in mounts.go, I can pass in the `coreAgent` and `processAgent` right?

Reply:
Yep!